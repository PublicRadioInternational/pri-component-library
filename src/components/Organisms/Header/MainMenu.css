/* import colors... */
@value colors: "../../00_global/colors.css";
@value whiteDark, gray, grayLighter, grayDark, orange, orangeDark, blue, yellow, green from colors;

/* import fonts... */
@value fonts: "../../00_global/fonts.css";
@value heading from fonts;

/* import breakpoints */
@value min as bp-min from "../../00_global/breakpoints.css";

.wrapper {
  background-color: gray;
  flex-flow: column;
  min-height: 100vh;
  position: relative;
}

@media bp-min {
  .wrapper {
    display: flex;
    overflow-y: auto;
  }
}

.header {
  border-top: 3px solid orange;
  flex: 0 0 auto;
  padding: 15px 15px 20px;
  color: grayLighter;
}

.closeMenu {
  align-items: center;
  background: none;
  border: none;
  color: grayLighter;
  cursor: pointer;
  display: flex;
  float: right;
  font-size: 0.95rem;
  padding: 0;
}

.inlineLink {
  height: 20px;
  margin: 0;
  width: 20px;
}

.headerText {
  clear: both;
  padding-top: 20px;
}

.headerLogo {
  width: 36%;
}

.maxWidth {
  max-width: 100%;
}

/* Buttons */
.btnGroup {
  margin: 14px 0 2px;
}

@media bp-min {
  display: flex;
}

.btnHeaderLeft,
.btnHeaderMiddle,
.btnHeaderRight {
  display: block;
  font: normal 0.875rem/1.5 heading;
  letter-spacing: 0;
  margin-bottom: 0.3rem;
  padding: 0.275rem 0.5rem;
  text-transform: none;
}

@media bp-min {
  .btnHeaderLeft,
  .btnHeaderMiddle,
  .btnHeaderRight {
    display: inline-block;
    padding: 0.475rem 0.5rem;
  }

  .btnHeaderLeft {
    border-bottom-right-radius: 0;
    border-top-right-radius: 0;
  }

  .btnHeaderMiddle {
    border-left: 1px solid orangeDark;
    border-radius: 0;
  }

  .btnHeaderRight {
    border-bottom-left-radius: 0;
    border-top-left-radius: 0;
    border-left: 1px solid orangeDark;
  }
}

/* Accordion Nav */
.accordionMenu {
  background: whiteDark;
  flex: 1 1 auto;
  padding: 10px;
}

/* Social Links */
.socialMenu {
  background-color: gray;
  display: flex;
  flex: 0 1 57px;
  justify-content: space-between;
}

.socialMenuItem {
  display: inline-block;
  overflow: hidden;
}

.socialMenuLink {
  color: gray;
  display: block;
  padding: 1rem;
}

.socialMenuLinkFB:hover {
  background-color: #5179bb;
  color: #5179bb;
}

.socialMenuLinkTW:hover {
  background-color: #1da1f2;
  color: #1da1f2;
}

.socialMenuLinkEM:hover {
  background-color: orange;
}

.socialMenuLinkRSS:hover {
  background-color: green;
  color: green;
}

.socialMenuLinkLogin:hover {
  background-color: blue;
}

.socialMenuIcon {
  background-color: #ddd;
  border-radius: 50px;
  height: 16px;
  padding: 5px 3px 5px 1px;
  width: 21px;
}

.socialMenuIconFull {
  background: none;
  border-radius: 0;
  color: #ddd;
  height: 25px;
  padding: 0;
  width: 25px;
}

.socialMenuIconRSS {
  height: 23px;
  padding: 2px 2px 2px 3px;
  width: 22px;
}

.socialMenuIconFullLg {
  height: 37px;
  width: 33px;
}
