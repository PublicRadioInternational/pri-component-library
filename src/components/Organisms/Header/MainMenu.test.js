/**
 * @file MainMenu.test.js
 * Contains tests for MainMenu.component.js.
 */

import React from 'react';
import renderer from 'react-test-renderer';

import MainMenu from './MainMenu.component';

describe('<MainMenu />', () => {
  it('Matches the Main Menu snapshot', () => {
    const component = renderer.create(<MainMenu />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
