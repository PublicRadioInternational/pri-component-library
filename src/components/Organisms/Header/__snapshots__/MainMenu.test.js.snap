// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu /> Matches the Main Menu snapshot 1`] = `
<div
  className="wrapper"
>
  <div
    className="search"
  >
    <form
      className="searchForm"
      method="POST"
      onSubmit={[Function]}
    >
      <label
        className="searchLabel"
        htmlFor="search__field"
      >
        Search
      </label>
      <input
        className="searchInput"
        id="search__field"
        onChange={[Function]}
        type="text"
      />
      <button
        className="searchBtn"
        type="submit"
      >
        <svg
          aria-hidden={null}
          aria-label={null}
          aria-labelledby={null}
          className="svg searchIcon"
          fill="currentcolor"
          height={28}
          version={1.1}
          viewBox="0 0 28 28"
          width={28}
        >
          <path
            d="M18 13c0-3.859-3.141-7-7-7s-7 3.141-7 7 3.141 7 7 7 7-3.141 7-7zM26 26c0 1.094-0.906 2-2 2-0.531 0-1.047-0.219-1.406-0.594l-5.359-5.344c-1.828 1.266-4.016 1.937-6.234 1.937-6.078 0-11-4.922-11-11s4.922-11 11-11 11 4.922 11 11c0 2.219-0.672 4.406-1.937 6.234l5.359 5.359c0.359 0.359 0.578 0.875 0.578 1.406z"
          />
        </svg>
      </button>
    </form>
  </div>
</div>
`;
